### PR TITLE
[BENCH-579] Add option to force delete AWS notebooks

### PR DESCRIPTION
**Pull request by dexamundsen**
- Notebooks can only be deleted when they are not in service (i.e. stopped)
- Expectation is. that user shall stop the notebook before calling delete
- There may be situations where caller wants to internally stop the notebook by default when delete is requested. Example Test cleanup
- This PR adds a deleteOption to the API to request force deletion

Testing -
```
create a workspace with - 2 notebooks and 2 storage folders
delete 1 storage folder and 1 stopped notebook directly - successful
delete workspace directly w/o stopping notebook - successful (internally stopped notebook before deletion)
```
log
```
2023-05-25 11:02:14.561   INFO 98159 --- [ool-8-thread-14] b.terra.workspace.common.utils.AwsUtils  : Stopping notebook instance dex-note-525-3
2023-05-25 11:02:14.970   INFO 98159 --- [ool-8-thread-14] b.terra.workspace.common.utils.AwsUtils  : Waiting on notebook with name dex-note-525-3, desired status Stopped.
…
2023-05-25 11:04:46.102   INFO 98159 --- [ool-8-thread-14] b.terra.workspace.common.utils.AwsUtils  : Deleting notebook instance dex-note-525-3
2023-05-25 11:04:47.108   INFO 98159 --- [ool-8-thread-14] b.terra.workspace.common.utils.AwsUtils  : Waiting on notebook with name dex-note-525-3, desired status Deleting.
…
```

### CODE REVIEW

**Comment by ddietterich** [comment]
What is the delete option for?
Why would you want to expose it through the API?

**Comment by rogerwangcs** [comment]
I don't think we need deleteOptions to be exposed on the resource deletion endpoint, we just need the workspace deletion flight to force delete the sagemaker notebooks.

**Comment by dexamundsen** [comment]
> I don't think we need deleteOptions to be exposed on the resource deletion endpoint, we just need the workspace deletion flight to force delete the sagemaker notebooks.

Makes sense, changing

**Comment by dexamundsen** [comment]
> What is the delete option for? Why would you want to expose it through the API?

Removed, not exposed

File: service/src/main/java/bio/terra/workspace/service/workspace/flight/cloud/aws/DeleteControlledAwsResourcesStep.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ public StepResult doStep(FlightContext flightContext)
     // Delete all resources
     for (ControlledResource resource : controlledResourceList) {
       controlledResourceService.deleteControlledResourceSync(
-          workspaceUuid, resource.getResourceId(), userRequest);
+          workspaceUuid, resource.getResourceId(), true, userRequest);
```
Comment (rogerwangcs) [inline service/src/main/java/bio/terra/workspace/service/workspace/flight/cloud/aws/DeleteControlledAwsResourcesStep.java:51]: nit `/* forceDelete= */`?

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/aws/sageMakerNotebook/StopAwsSageMakerNotebookStep.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                    "AWS SageMaker Notebook resource %s, being deleted.",
+                    resource.getResourceId(), notebookStatus));
+          }
+          // else: being deleted
+        }
+        case PENDING, UPDATING, UNKNOWN_TO_SDK_VERSION -> throw new ApiException(
+            String.format(
+                "Cannot stop AWS SageMaker Notebook resource %s, status %s.",
+                resource.getResourceId(), notebookStatus));
+        case STOPPED, FAILED -> {
```
Comment (rogerwangcs) [inline service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/aws/sageMakerNotebook/StopAwsSageMakerNotebookStep.java:67]: Does `FAILED` mean the status call failed or the notebook instance is in a failed state? Is it possible that we can end up with orphaned notebooks?

**Review by rogerwangcs — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/service/workspace/flight/cloud/aws/DeleteControlledAwsResourcesStep.java (hunk 1 above)
Comment (dexamundsen) [reply service/src/main/java/bio/terra/workspace/service/workspace/flight/cloud/aws/DeleteControlledAwsResourcesStep.java:51]: done

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/aws/sageMakerNotebook/StopAwsSageMakerNotebookStep.java (hunk 2 above)
Comment (dexamundsen) [reply service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/aws/sageMakerNotebook/StopAwsSageMakerNotebookStep.java:67]: FAILED -> the notebook instance is in failed state i.e. not in service. Hence the deletion can go on if needed
No, this wont lead to orphaned notebooks.

File: service/src/main/java/bio/terra/workspace/common/utils/AwsUtils.java [hunk 3]
```diff
@@ -701,6 +692,40 @@ public static NotebookInstanceStatus getSageMakerNotebookStatus(
     }
   }
 
+  /**
+   * Start a AWS SageMaker Notebook
+   *
+   * @param awsCredentialsProvider {@link AwsCredentialsProvider}
+   * @param notebookResource {@link ControlledAwsSageMakerNotebookResource}
+   */
+  public static void startSageMakerNotebook(
```
Comment (jmczerk) [inline service/src/main/java/bio/terra/workspace/common/utils/AwsUtils.java:701]: Is this used for anything?

**Review by jmczerk — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/common/utils/AwsUtils.java (hunk 3 above)
Comment (dexamundsen) [reply service/src/main/java/bio/terra/workspace/common/utils/AwsUtils.java:701]: yes, in undo of STOP step. 
(This too should be leverage from CRL later)

**Pull request merged**